File: apps/go/domain/type_/fizz_buzz_type_02.go
package type_

import (
	"strconv"

	"github.com/k2works/getting-started-tdd/apps/go/domain/model"
)

// FizzBuzzType02 は数値のみを返します。
type FizzBuzzType02 struct{ fizzBuzzTypeBase }

func (f FizzBuzzType02) Generate(number int) model.FizzBuzzValue {
	return model.NewFizzBuzzValue(number, strconv.Itoa(number))
}

File: apps/java/src/main/java/tdd/fizzbuzz/domain/type/FizzBuzzType02.java
package tdd.fizzbuzz.domain.type;

import tdd.fizzbuzz.domain.model.FizzBuzzValue;

public final class FizzBuzzType02 extends FizzBuzzType {

    @Override
    public FizzBuzzValue generate(int number) {
        return new FizzBuzzValue(number, Integer.toString(number));
    }
}
